### PR TITLE
User Authentication and Login

### PR DESCRIPTION
**Pull request by PanuphongK**

### CODE REVIEW

File: src/auth/dto/login.dto.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+export class LoginDto{
+    @IsNotEmpty()
+    @IsString({ message: 'Please enter correct username'})
+    readonly username: string
+
+    @IsNotEmpty()
+    @IsString()
+    @MinLength(6)
+    readonly password: string
```
Comment (EyeOfCode) [inline src/auth/dto/login.dto.ts]: You don't need to use `readonly`. you can use `password: string`

File: src/auth/schemas/user.schema.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+
+@Schema({
+    timestamps: true
+})
+export class User {
+    @Prop()
+    name: string
+
+    @Prop({ unique: [ true, 'Duplicate username entered']})
```
Comment (EyeOfCode) [inline src/auth/schemas/user.schema.ts]: Can it really be used? i not sure this is can be used. can you test again please? i think you can use `{ unique: true, message: "User duplication error" }` if it not work.

**Review by aecuto — CHANGES_REQUESTED** [review]
please rename PRs that make more readable dont use `feat:`

File: src/auth/auth.controller.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    @Post('/signup')
+    signUp(@Body() signUpDto: SignUpDto): Promise<{token: string}> {
+        return this.authService.signUp(signUpDto)
+    }
+
+    @Get('/login')
+    login(@Body() loginDto: LoginDto): Promise<{token: string}> {
+        return this.authService.login(loginDto)
+    }
+}
```
Comment (aecuto) [inline src/auth/auth.controller.ts:19]: add GET `/me` api

**Review by aecuto — CHANGES_REQUESTED** [review]

**Pull request closed without merging**